Added coding to dashboard pages

vicbro00 opened the pull request:
Added html and css to dashboard for admin, teacher and student.

### What does this PR do?
Adds coding to three pages.

### Description of Task to be completed?
Creates profile pages

### How should this be manually tested?
Open live servers and ensure everything works and is visual

### Any background context you want to provide?
I created the hamburger menus, but can't figure out how to make them appear and dissappear so I have left them hidden.

vicbro00 committed:
Added coding to dashboard pages

Added html and css to dashboard for admin, teacher and student.

kristinatonnessen commented on templates/dashboard-admin.html:
```diff
@@ -0,0 +1,103 @@
+    <title>Admin Profile Page</title>
+</head>
+<body>
+    <header>
```
Remove header, this will be coded by other teams and added when finished

Anne-Zwift replied:
Header is removed.

kristinatonnessen commented on templates/dashboard-admin.html:
```diff
@@ -0,0 +1,103 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta name="description" content="Admin profile page for Edugate webpage.">
+    <link rel="stylesheet" href="./output.css">
+    <link rel="stylesheet" href="./css/style.css">
```
update to the correct stylesheet direction

Anne-Zwift replied:
I've made a PR for this one.

kristinatonnessen commented on templates/dashboard-admin.html:
```diff
@@ -0,0 +1,103 @@
+    </header>
+
+    <main>
+        <!--Hamburger menu-->
```
remove burger menu

Anne-Zwift replied:
Do you mean delete all of this code? 
![image](https://github.com/user-attachments/assets/06bd9129-d541-496c-95d0-4585a6e22027)

kristinatonnessen replied:
yes:)

kristinatonnessen commented on templates/dashboard-admin.html:
```diff
@@ -0,0 +1,103 @@
+                <img class="cursor-pointer w-12" src="/public/icons/logout.png" alt="Picture of a red log out icon">
+            </div>
+
+            <!--Mobile view-->
```
everything in main is hidden om medium screen view

Anne-Zwift replied:
The main is now visible and updated #277

kristinatonnessen commented on templates/dashboard-admin.html:
```diff
@@ -0,0 +1,103 @@
+
+    <footer class="mt-8 w-full flex">
+        <img class="w-full max-w-md mx-auto" src="/public/img/circle-footer.png" alt="Picture of a blue circle overlapping a green circle">
+    </footer>
```
change footer to the updated version for both mobile and desktop

Anne-Zwift replied:
@kristinatonnessen @Janicke92  regarding the footer. I watched the Figma illustration and can't find that they use this code in the footer. I only see the circles. Should I use <p>&copy; 2025 Edugate. All Rights Reserved.</p> in the footer?

kristinatonnessen replied:
yeah, i dont know why, but that can be removed and add the footer from the design :)

kristinatonnessen commented on templates/dashboard-student.html:
```diff
@@ -0,0 +1,70 @@
+    <meta name="description" content="Student profile page for Edugate webpage.">
+    <link rel="stylesheet" href="/student-profile-mobile.css">
+    <link rel="stylesheet" href="./css/style.css">
+    <link href="https://cdn.jsdelivr.net/npm/tailwindcss@2.2.19/dist/tailwind.min.css" rel="stylesheet">
```
Remove both stylesheets and add the correct/updated direction

Anne-Zwift replied:
Will it be linked to two different stylesheets?

Anne-Zwift replied:
I made a PR removing the two stylesheet directions and adding the correct one.

kristinatonnessen commented on templates/dashboard-student.html:
```diff
@@ -0,0 +1,70 @@
+        </div>
+    </header>
+
+    <main>
```
all the same notes as from the admin page

Anne-Zwift replied:
Do you mean remove the header?

kristinatonnessen replied:
yeah, you can add the official header instead or leave it open for now and all headers and footers can be added in a later issue

kristinatonnessen commented on templates/dashboard-teacher.html:
```diff
@@ -0,0 +1,96 @@
+    <meta name="description" content="Teacher profile page for Edugate webpage.">
+    <link rel="stylesheet" href="/student-profile-mobile.css">
+    <link rel="stylesheet" href="./css/style.css">
+    <link href="https://cdn.jsdelivr.net/npm/tailwindcss@2.2.19/dist/tailwind.min.css" rel="stylesheet">
```
same notes as from the admin page

Anne-Zwift replied:
Not sure of what you mean please elaborate.

Anne-Zwift replied:
Removing the "/student-profile-mobile.css" and the "./css/style.css". Adding the correct direction.

kristinatonnessen reviewed (changes requested):
and remove burger menu / student initials as those are supposed to be changed out as who is logged in and need to be coded and not as an img?

Anne-Zwift committed:
Update dashboard-admin.html

changed the stylesheet link to: <link rel="stylesheet" href="../dist/output.css">

Anne-Zwift committed:
Update dashboard-admin.html

<!--header is removed and being replaced-->

Janicke92 committed:
Merge pull request #268 from NoroffFEU/Anne-Zwift-patch-1

Update dashboard-admin.html

Janicke92 committed:
Merge pull request #269 from NoroffFEU/Anne-Zwift-patch-2

Update dashboard-admin.html

kristinatonnessen reviewed (dismissed):
nice :D

Anne-Zwift committed:
Update dashboard-student.html

I removed the two stylesheet links and replaced it with "..dist/output.css link.

Janicke92 committed:
Merge pull request #275 from NoroffFEU/Anne-Zwift-patch-3

Update dashboard-student.html

Anne-Zwift committed:
Update dashboard-teacher.html

Adding <link rel="stylesheet" href="../dist/output.css"> and removing the other directions.

Janicke92 committed:
Merge pull request #278 from NoroffFEU/Anne-Zwift-patch-5

Update dashboard-teacher.html

Anne-Zwift committed:
Update dashboard-admin.html

Deleting the code for hamburger menu. #255

Anne-Zwift committed:
Update dashboard-student.html

Removing the header code.
#255

Anne-Zwift committed:
Update dashboard-admin.html

Removing hidden from desktop view

Anne-Zwift committed:
student initials image

Anne-Zwift committed:
Update dashboard-admin.html

Added direction to image "student initials".

Anne-Zwift committed:
student-initials-image-mobile

Anne-Zwift committed:
Update dashboard-admin.html

Added direction to image in mobile view.

Anne-Zwift committed:
Update dashboard-admin.html

added lg: hidden to desktop view.

Anne-Zwift committed:
Update dashboard-admin.html

Removing an extra "grid" word in the code line 43, desktop view.

Anne-Zwift committed:
Update dashboard-admin.html

Undoing last line 43, lg:hidden

Janicke92 committed:
Merge pull request #281 from NoroffFEU/Anne-Zwift-patch-8

Update dashboard-student.html

Janicke92 committed:
Merge branch 'develop' into Dashboard-Admin

Anne-Zwift committed:
Update dashboard-admin.html

Fixed view for all sizes according to the Figma file.

Anne-Zwift committed:
Merge branch 'develop' of https://github.com/NoroffFEU/edu-gate into develop

Anne-Zwift committed:
save changes to dashboard-admin.html

Anne-Zwift committed:
adding student initials png

Anne-Zwift committed:
completed the dashboard teacher page

Anne-Zwift committed:
added text-sizes to h1

Anne-Zwift committed:
Update dashboard-admin.html

added text-size to h1

kristinatonnessen committed:
Merge pull request #280 from NoroffFEU/Anne-Zwift-patch-7

Update dashboard-admin.html

kristinatonnessen committed:
Merge branch 'develop' into Dashboard-Admin

kristinatonnessen committed:
fixed styles updates from manage user

kristinatonnessen committed:
updated footer in manage user

kristinatonnessen commented on templates/dashboard-student.html:
```diff
@@ -0,0 +1,57 @@
+    <meta name="description" content="Student profile page for Edugate webpage.">
+    <link rel="stylesheet" href="../dist/output.css">
+    <link href="https://cdn.jsdelivr.net/npm/tailwindcss@2.2.19/dist/tailwind.min.css" rel="stylesheet">
+    <title>Student Profile Page</title>
```
change name to dashboard

Anne-Zwift replied:
@kristinatonnessen changed title to "Student Dashboard Page"

Anne-Zwift committed:
Update dashboard-admin.html

title changed to Admin Dashboard page

Anne-Zwift committed:
Update dashboard-teacher.html

added h4 and profile icon to the mobile view

Anne-Zwift committed:
Update dashboard-student.html

changed title to "Student Dashboard Page"

Anne-Zwift committed:
Update dashboard-admin.html

added h4 and profile icon to the dashboard admin page in mobile view.

kristinatonnessen committed:
Merge branch 'Dashboard-Admin' into dashboard-teacher-html

kristinatonnessen committed:
Merge pull request #298 from NoroffFEU/dashboard-teacher-html

Dashboard teacher html

kristinatonnessen commented on templates/dashboard-student.html:
```diff
@@ -0,0 +1,57 @@
+                <h4>Log out</h4>
+            </div>
+
+            <!--Desktop view-->
```
The view is still hidden in medium size, maybe the code got lost somewhere?

https://github.com/user-attachments/assets/b50d7903-c5fd-43ee-9981-934d416ffc05

Anne-Zwift replied:
fixed

kristinatonnessen commented on templates/dashboard-student.html:
```diff
@@ -0,0 +1,57 @@
+
+            <!--Mobile view-->
+            <div class="md:hidden grid gap-4 justify-items-center text-center">
+                <h1 class="font-extrabold">Dashboard</h1>
```
change size of text in desktop

Anne-Zwift replied:
text-3xl is added to h1 in desktop view

kristinatonnessen commented on templates/dashboard-student.html:
```diff
@@ -0,0 +1,57 @@
+            <!--Mobile view-->
+            <div class="md:hidden grid gap-4 justify-items-center text-center">
+                <h1 class="font-extrabold">Dashboard</h1>
+                <img src="/public/icons/Student-initials.png" alt="Picture of a green circle with the initials JB">
```
desktop view needs the bigger icon

Anne-Zwift replied:
Update dashboard-student: added student-initials icon to desktop view.

kristinatonnessen commented on templates/dashboard-teacher.html:
```diff
@@ -0,0 +1,77 @@
+                <div class="grid grid-cols-3 gap-12 m-4 pl-24">
+                    <div>
+                        <img class="cursor-pointer" src="/public/icons/top-students.png" alt="Picture of a green student with a hat icon">
+                        <h4>Top students</h4>
```
text-center for desktop view for icon text 
![Skjermbilde 2025-01-31 kl  12 57 38](https://github.com/user-attachments/assets/dcc992e3-fcc0-4bbd-9299-f781c6114f27)

Anne-Zwift replied:
added h4 to center

kristinatonnessen reviewed (changes requested):
Think some coding for student desktop was lost in the different pr's or commits, just add those and some small changes and we're good to go!

Anne-Zwift committed:
Update dashboard-student.html

Updating dashboard- student: making desktop view visible

Anne-Zwift committed:
Update dashboard-student.html

Udate dashboard-student: adding text-size h1 3xl to desktop view

Anne-Zwift committed:
Update dashboard-student.html

Update dashboard-student: added student-initials icon to desktop view.

Anne-Zwift committed:
Update dashboard-teacher.html

update dashboard-teacher: making h4 center

Anne-Zwift commented:
@kristinatonnessen I think I've done all requests on this one. Should I close it?

kristinatonnessen commented:
@Anne-Zwift no don't close, all the commits come back to this PR. This one needs to be reviewed and merged 😃

Anne-Zwift commented:
@kristinatonnessen I see I’ll leave it open 😁

kristinatonnessen committed:
Merge branch 'develop' into Dashboard-Admin
